cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/SpendingMonthPage.java

```diff
@@ -0,0 +1,164 @@
+
+    /**
+     * Initializes the driver and sets an implicit wait 
+     */
+    public SpendingMonthPage(WebDriver driver){
+        super(driver);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        PageFactory.initElements(driver, this);
+    }
+
+    // Helper method to wait for element visibility
+    private WebElement waitForElement(WebElement element, int timeoutInSeconds) {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(timeoutInSeconds));
+        return wait.until(ExpectedConditions.visibilityOf(element));
+    }
+
+    public void clickTab() {
+        waitForElement(spendingsTab, 10).click();
+        try {
+            Thread.sleep(500);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public List<Component> getChildComponents() {
+        return Arrays.asList(dashboardNavbar);
+    }
+
+    @Override
+    public Component getChildComponent(String name) {
+        switch (name) {
+            case CMP_DASHBOARDNAVBAR_NAME:
+                return dashboardNavbar;
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Returns a List of all WebElements within this component. Searches sub-components as well.
+     */
+    @Override
+    public List<WebElement> getWebElements() {
+        ArrayList<WebElement> webElements = new ArrayList<WebElement>();
+        webElements.add(btnBackToSpending);
+
+        for(Component component : getChildComponents()){
+            webElements.addAll(component.getWebElements());
+        }
+
+        return webElements;
+    }
+    
+    /**
+     * Returns a WebElement given a name. Searches sub-components as well.
+     * return WebElement requested, or null if not found.
+     */
+    @Override
+    public WebElement getWebElement(String name) {
+        switch (name) {
+            case BTN_BACKTOANNUALSPENDING_NAME:
+                return btnBackToSpending;
+            default:
+                for (Component component : getChildComponents()) {
+                    WebElement webElement = component.getWebElement(name);
+                    if(webElement != null) return webElement;
+                }
+                break;
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns a list of all buttons directly under this component.
+     */
+    @Override
+    public List<WebElement> getButtons() {
+        return Arrays.asList(btnBackToSpending);
+    }
+
+    /**
+     * Clicks a button given a name.
+     */
+    @Override
+    public void clickButton(String name) {
+        WebElement button = getWebElement(name);
+        
+        if(button != null) {
+            waitForElement(button, 10).click();
+        }
+        else {
+            throw new IllegalArgumentException("Button '" + name + "' does not exist.");
+        }
+
+        // give time for page to load
+        try {
+            Thread.sleep(2000);
```

Comment:
Low impact: Using a 2 second sleep could drastically increase testing times over the course of a full project.
Suggestion: Use the waitForElement method for an element that should always be on the page.

Reply:
Updated, all occurrences of try{sleep} are removed, only waitForElements being used in place